servicetalk-data-jackson-jersey/src/main/java/io/servicetalk/data/jackson/jersey/JacksonSerializationExceptionMapper.java

```diff
@@ -34,6 +35,20 @@ public Response toResponse(final SerializationException e) {
     }
 
     private static boolean isDueToBadUserData(final SerializationException e) {
-        return e.getCause() instanceof JsonMappingException || e.getCause() instanceof JsonParseException;
+        if (e.getCause() instanceof JsonParseException) {
+            return true;
+        }
+
+        if (e.getCause() instanceof JsonMappingException) {
+            // Mapping exceptions can happen from both parsing and writing. We attempt to
+            // distinguish between the two by the processor type which should be either a
+            // JsonGenerator or a JsonParser.
+            JsonMappingException ex = (JsonMappingException) e.getCause();
+            return !(ex.getProcessor() instanceof JsonGenerator);
```

Comment:
is it possible `ex == null`?

Reply:
It's possible and in the case that it is the existing behavior is preserved. I added a test for that.

Reply:
I'm sorry, I think I misread your question. I don't think `ex` can be null or else it wouldn't have passed the `instancesof` check since null is not an instance of anything.